app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Katello
+  class Api::Rhsm::CandlepinDynflowProxyController < Api::V2::ApiController
+    include Katello::Authentication::ClientAuthentication
+    include ForemanTasks::Triggers
+    include AbstractController::Callbacks
+
+    before_action :find_host, :only => [:upload_package_profile]
+    before_action :authorize_client_or_user, :only => [:upload_package_profile]
+
+    #api :PUT, "/consumers/:id/packages", N_("Update installed packages")
+    #api :PUT, "/consumers/:id/profile", N_("Update installed packages")
+    #param :id, String, :desc => N_("UUID of the consumer"), :required => true
+    def upload_package_profile
+      User.as_anonymous_admin do
+        async_task(::Actions::Katello::Host::UploadPackageProfileWrapper, @host, request.body.string)
+      end
+      [
+        200,
+        { "Content-Type" => "application/json" },
+        Resources::Candlepin::Consumer.get(@host.subscription_facet.uuid)
+      ]
+    end
```

Comment:
can you think of anyway to write a test to test that would test that the body isn't parsed?  the code handling that seems a bit fragile so it would be nice if we could test that.  Not a pure blocker to merging though.